### ISSUE TITLE
Two-column rendering?

### ISSUE THREAD

**Issue by orloffm** [asciidoctor/asciidoctor-pdf#327]
Sorry for using "issues" for this, but is it possible to create a two-column "scientific" layout for some text?

I need to create a huge table of key words and their appearances. Not sure even to how approach this.

**Comment by mojavelinux** [comment]
Prawn does support columns, but we have not implemented in Asciidoctor PDF for a few reasons. First, it would introduce even more challenges to calculating and arranging blocks and we haven't solved the problems with a single column yet. Second, two columns seems to me more of a legacy layout from the print days and less needed in modern documents. But I'm open to listening other views.

If you need two columns to achieve side-by-side content, then I would recommend using a table. Keep in mind that we don't yet have support for AsciiDoc content cells, so that presents some challenges. But that fix is in development and is scheduled for the beta1 release.

**Comment by chad-loder** [comment]
I would like to see two-column layout with automatic flow. This is very popular with legal contracts, and certain well-known management consulting companies produce easy-to-recognize two-column layouts. This is quite doable in Pandoc -> CSS - > PDF and it would be lovely if asciidoctor-pdf could support it as well.

**Comment by mojavelinux** [comment]
This is a lot of work and I don't have time to commit to it. If someone in the community is willing to take it on, I would certainly sign-off on the enhancement.

**Comment by ahus1** [comment]
There is another project in the Asciidoctor ecosystem called "Asciidoctor.js PDF" that supports multi-column layouts. This links to an example: https://github.com/Mogztter/asciidoctor-pdf.js/tree/master/examples/cheat-sheet

TL;DR: styling is done using CSS, layout of the content plus PDF generation is done through headless chrome/puppeteer.

**Comment by mojavelinux** [comment]
That's correct. I'm probably not going to pursue two-column rendering in Asciidoctor PDF. If someone is willing to implement it, and can demonstrate it passes all the tests, I'd strongly consider merging it in. But there's a limit to what we can do with layouts in Asciidoctor PDF and is the future asciidoctor-pdf.js is going to fill.

**Comment by mojavelinux** [comment]
Believe it or not...

Thanks to improvements made to the code for Asciidoctor PDF 2, the foundation is now in place to support the two-column layout. For now, I have documented how to apply a multi-column layout to the document body using an extended converter. See https://docs.asciidoctor.org/pdf-converter/latest/extend/use-cases/#multiple-columns

We're working on adding built-in support for multi-column layout out of the box in a future release.

**Comment by chad-loder** [comment]
Neato! I will check it out!

**Comment by mojavelinux** [comment]
> We're working on adding built-in support for multi-column layout out of the box in a future release.

This will be included in the 2.1.0 release.
